R/substitution.R

```diff
@@ -0,0 +1,609 @@
+#'   large enough for replacement.
+#' @param large_pattern The pattern text to be used in place of the suitably
+#'   large values in the rendered table.
+#'
+#' @return An object of class `gt_tbl`.
+#'
+#' @section Examples:
+#'
+#' Let's generate a simple, single-column tibble that contains an assortment of
+#' values that could potentially undergo some substitution.
+#'
+#' ```{r}
+#' tbl <- dplyr::tibble(num = c(0, NA, 10^(8:14)))
+#'
+#' tbl
+#' ```
+#'
+#' The `tbl` contains a variety of smaller numbers and some might be small
+#' enough to reformat with a threshold value. With `sub_small_vals()` we can
+#' do just that:
+#'
+#' ```r
+#' tbl %>%
+#'   gt() %>%
+#'   fmt_number(columns = num) %>%
+#'   sub_large_vals()
+#' ```
+#'
+#' \if{html}{\out{
+#' `r man_get_image_tag(file = "man_sub_large_vals_1.png")`
+#' }}
+#'
+#' Large negative values can also be handled but they are handled specially
+#' by the `sign` parameter. Setting that to `"-"` will format only the large
+#' values that are negative. Notice that with the default `large_pattern`
+#' value of `">={x}"` the `">="` is automatically changed to `"<="`.
+#'
+#' ```r
+#' tbl %>%
+#'   dplyr::mutate(num = -num) %>%
+#'   gt() %>%
+#'   fmt_number(columns = num) %>%
+#'   sub_large_vals(sign = "-")
+#' ```
+#'
+#' \if{html}{\out{
+#' `r man_get_image_tag(file = "man_sub_large_vals_2.png")`
+#' }}
+#'
+#' You don't have to settle with the default `threshold` value or the default
+#' replacement pattern (in `large_pattern`). This can be changed and the
+#' `"{x}"` in `large_pattern` (which uses the `threshold` value) can even be
+#' omitted.
+#'
+#' ```r
+#' tbl %>%
+#'   gt() %>%
+#'   fmt_number(columns = num) %>%
+#'   sub_large_vals(
+#'     threshold = 5E10,
+#'     large_pattern = "hugemongous"
+#'   )
+#' ```
+#'
+#' \if{html}{\out{
+#' `r man_get_image_tag(file = "man_sub_large_vals_3.png")`
+#' }}
+#'
+#' @family Format Data
+#' @section Function ID:
+#' 3-19
+#'
+#' @import rlang
+#' @export
+sub_large_vals <- function(
+    data,
+    columns = everything(),
+    rows = everything(),
+    threshold = 1E12,
+    large_pattern = ">={x}",
+    sign = "+"
+) {
+
+  # Perform input object validation
+  stop_if_not_gt(data = data)
+
+  # Check that the `sign` value is an acceptable value
+  check_sub_fn_sign(sign = sign)
+
+  if (sign == "+") {
+    op_fn <- `>=`
+  } else {
+    op_fn <- `<=`
+  }
+
+  # Pass `data`, `columns`, `rows`, and the formatting
+  # functions (as a function list) to `fmt()`
+  fmt(
+    data = data,
+    columns = {{ columns }},
+    rows = {{ rows }},
+    prepend = TRUE,
+    fns = list(
+      # Any values of `x` that are above the threshold will be processed
+      # according to the `large_pattern`, the `threshold` value (interacts with
+      # the `large_pattern`, and the sign (changes the default `large_pattern`))
+      html = function(x) {
+        ifelse(
+          is.numeric(x) &
```

Comment:
This `&` doesn't short-circuit. Let's perform a separate check for `is.numeric(x)` and return `rep_len(NA, length(x))`

Reply:
Now done. Thanks for this!